src/main/kotlin/no/nav/syfo/service/SenderFacade.kt

```diff
@@ -76,29 +83,45 @@ class SenderFacade(
         content: String,
         url: URL? = null,
         varselHendelse: ArbeidstakerHendelse,
-        meldingType: BrukernotifikasjonKafkaProducer.MeldingType? = BrukernotifikasjonKafkaProducer.MeldingType.BESKJED,
+        varseltype: InternalBrukernotifikasjonType,
         eksternVarsling: Boolean = true,
         smsContent: String? = null,
     ) {
-        var isSendingSucceed = true
         try {
-            brukernotifikasjonerService.sendVarsel(uuid, mottakerFnr, content, url, meldingType, eksternVarsling, smsContent)
+            when (varseltype) {
+                InternalBrukernotifikasjonType.BESKJED -> brukernotifikasjonerService.sendBeskjed(
+                    uuid = uuid,
+                    mottakerFnr = mottakerFnr,
+                    content = content,
+                    url = url,
+                    eksternVarsling = eksternVarsling
+                )
+
+                InternalBrukernotifikasjonType.OPPGAVE -> brukernotifikasjonerService.sendOppgave(
+                    uuid = uuid,
+                    mottakerFnr = mottakerFnr,
+                    content = content,
+                    url = url,
+                    smsContent = smsContent
+                )
+
+                InternalBrukernotifikasjonType.DONE -> brukernotifikasjonerService.ferdigstillVarsel(uuid = uuid)
+            }
+            lagreUtsendtArbeidstakerVarsel(
+                kanal = BRUKERNOTIFIKASJON,
+                varselHendelse = varselHendelse,
+                eksternReferanse = uuid
+            )
         } catch (e: Exception) {
             log.warn("Error while sending varsel to BRUKERNOTIFIKASJON: ${e.message}")
-            isSendingSucceed = false
             lagreIkkeUtsendtArbeidstakerVarsel(
                 kanal = BRUKERNOTIFIKASJON,
                 varselHendelse = varselHendelse,
                 eksternReferanse = uuid,
                 feilmelding = e.message,
-                journalpostId = null,
-                brukernotifikasjonerMeldingType = meldingType?.name
-                    ?: BrukernotifikasjonKafkaProducer.MeldingType.BESKJED.name,
+                journalpostId = null
```

Comment:
Jeg ser at du har tatt vekk `brukernotifikasjonerMeldingType`. Er det bevisst? Jeg tror tanken var at det skulle bli brukt til resending, men det er ikke implementert ennå.

Reply:
Ah ja stemmer. Jeg tok det vekk først fordi brukernotifikasjoner ikke lenger har DONE som en varseltype. Men fant til slutt ut at jeg lager InternalBrukernotifikasjonType som er lik som den gamle varseltype-greia, for å unngå scope creep med type mismatch.

Har glemt å legge tilbake den typen ved lagring av feil, fikser.

Reply:
Okei. Jeg synes egentlig det hadde vært bedre at man kalte en av ferdigstill-metodene i stedet for å sende type = DONE, men det er kanskje en større jobb.

Reply:
Nå har jeg forsøkt 2 ganger, og det ble bare ball :P Foreslår vi tar det i egen PR